# 10 行数のカウント
行数をカウントせよ．確認にはwcコマンドを用いよ．

In [2]:
# pandasを用いてデータを読み込む
import pandas as pd

/Users/ishimotoyuuta/.pyenv/versions/3.6.5/lib/python3.6/site-packages/pandas/compat/__init__.py:85: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)
/Users/ishimotoyuuta/.pyenv/versions/3.6.5/lib/python3.6/site-packages/pandas/compat/__init__.py:85: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)


In [3]:
# テキストデータから読み込み
data = pd.read_table('popular_names.txt', header=None)
pd.read_table('popular_names.txt', header=None)

0  1      2     3
0          Mary  F   7065  1880
1          Anna  F   2604  1880
2          Emma  F   2003  1880
3     Elizabeth  F   1939  1880
4        Minnie  F   1746  1880
...         ... ..    ...   ...
2775   Benjamin  M  13381  2018
2776     Elijah  M  12886  2018
2777      Lucas  M  12585  2018
2778      Mason  M  12435  2018
2779      Logan  M  12352  2018

[2780 rows x 4 columns]

In [24]:
# 解答
print(len(data))

2780


In [19]:
# 列数の取得も簡単にできる
print(len(data.columns))

# 行数や列数などの情報
print(data.info())

4
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2780 entries, 0 to 2779
Data columns (total 4 columns):
0    2780 non-null object
1    2780 non-null object
2    2780 non-null int64
3    2780 non-null int64
dtypes: int64(2), object(2)
memory usage: 87.0+ KB
None


## コマンドで確認

`wc -l popular_names.txt`で行数をカウントできるが結果は2779と1少なくなる．

これは`-l`オプションが改行の数を数えているため．
最後のデータの末尾にも改行を入れてやれば2780にはなる．

その他オプションにより，バイト数や文字数を確認できる．

cf) https://www.atmarkit.co.jp/ait/articles/1611/07/news026.html